zipline/finance/commission.py

```diff
@@ -69,6 +69,18 @@ def calculate(self, order, transaction):
         """
         raise NotImplementedError('calculate')
 
+    def __eq__(self, other):
+        return self.asdict() == other.asdict()
+
+    def __hash__(self):
+        return hash((
+            type(self),
+            tuple(sorted(iteritems(self.asdict())))
```

Comment:
Hmm yeah that's a good point. My main rationale for implementing it was simply that is was implemented for slippage as well https://github.com/quantopian/zipline/blob/master/zipline/finance/slippage.py#L168

I don't feel strongly either way, but I think commissions and slippage should both implement the `__hash__` or neither. Unless there's something special about slippage that reasonably restricts its attributes to all being hashable